### DISCUSSION TITLE
How do HomeSpan / HomeKit handle renames and reordering?

### DISCUSSION THREAD

**Discussion by DirtyHairy** [HomeSpan/HomeSpan#233] [Q&A] [answered]
Something that I have been wondering: Suppose I have multiple accessories on a bridge, and I now start reordering and / or renaming them. How do HomeSpan and HomeKit handle identity? By which properties is the identity of a accessory established, even though I have moved or renamed it?

**Comment by HomeSpan** [comment]
HomeKit keeps track of Accessories by their AID (Accessory ID number).  HomeSpan automatically assigns these as part of [SpanAccessory()](https://github.com/HomeSpan/HomeSpan/blob/master/docs/Reference.md#spanaccessoryuint32_t-aid), or you can assign numbers yourself and keep them constant even as you add/subtract accessories from the bridge.   To see this in practice, check out my [ProgrammableHub](https://github.com/HomeSpan/ProgrammableHub) example.

**Comment by DirtyHairy** [comment, marked as answer]
Thanks for the fast response. Ah-ha, the AID is what eluded me 😏 So, I guess I can get away by setting the bridge accessory to AID 1, and then am free to choose the AIDs of all subsequent accessories as I like, correct?

**Comment by DirtyHairy** [comment]
EDIT: Sorry, I should have looked closer at your example before posting, it seems that this is exactly what it is doing.
